src/commonMain/kotlin/exchange.dydx.abacus/validator/TriggerOrdersInputValidator.kt

```diff
@@ -341,15 +351,39 @@ internal class TriggerOrdersInputValidator(
     private fun validateCalculatedPricesPositive(
         triggerOrder: Map<String, Any>,
     ): List<Any>? {
+        val type = parser.asString(triggerOrder["type"])
         val triggerPrice = parser.asDouble(parser.value(triggerOrder, "price.triggerPrice"))
         val limitPrice = parser.asDouble(parser.value(triggerOrder, "price.limitPrice"))
+        val fields = if (type == "STOP_LIMIT" || type == "STOP_MARKET") {
+            var list: MutableList<String> = mutableListOf()
+            if (triggerPrice != null && triggerPrice <= 0) {
+                list.add(TriggerInputField.SL_TRIGGER_PRICE.rawValue)
+                list.add(TriggerInputField.SL_GAIN_LOSS.rawValue)
+            }
+            if (limitPrice != null && limitPrice <= 0) {
+                list.add(TriggerInputField.SL_LIMIT_PRICE.rawValue)
+            }
+            list
+        } else if (type == "TAKE_PROFIT" || type == "TAKE_PROFIT_MARKET") {
+            var list: MutableList<String> = mutableListOf()
+            if (triggerPrice != null && triggerPrice <= 0) {
+                list.add(TriggerInputField.TP_TRIGGER_PRICE.rawValue)
+                list.add(TriggerInputField.TP_GAIN_LOSS.rawValue)
+            }
+            if (limitPrice != null && limitPrice <= 0) {
+                list.add(TriggerInputField.TP_LIMIT_PRICE.rawValue)
```

Comment:
Instead of the `TriggerInputField` type, I wonder if, to be more in line with what `TradeInputDataValidator` [does](https://github.com/dydxprotocol/v4-abacus/blob/ed1bc7241131ce6fa3ebea97489c760aa1a2a06f/src/commonMain/kotlin/exchange.dydx.abacus/validator/trade/TradeInputDataValidator.kt#L333), we should be using things like `stopLossOrder.price.triggerPrice`, `stopLossOrder.price.limitPrice`, stopLossOrder.price.usdcDiff` etc instead?

Reply:
OK.. Will update.  But it's a just a string token, and once we migrate to static type it will just be an enum.